diff --git a/meson.build b/meson.build
index 1111111..2222222 100644
--- a/meson.build
+++ b/meson.build
@@ -42,12 +42,7 @@ if not have_bash
   error('bash >= 4.4.0 is required for pacman scripts.')
 endif

-bashcompletion = dependency('bash-completion', required : false)
-if bashcompletion.found()
-  BASHCOMPDIR = bashcompletion.get_variable(pkgconfig: 'completionsdir')
-else
-  BASHCOMPDIR = join_paths(DATAROOTDIR, 'bash-completion/completions')
-endif
+BASHCOMPDIR = join_paths(DATAROOTDIR, 'bash-completion/completions')

 if get_option('use-git-version')
   PACKAGE_VERSION = run_command(
