diff --git a/lib/libalpm/dload.c b/lib/libalpm/dload.c
index f2fa1a54..f5edb86d 100644
--- a/lib/libalpm/dload.c
+++ b/lib/libalpm/dload.c
@@ -1139,6 +1139,7 @@ int SYMEXPORT alpm_fetch_pkgurl(alpm_handle_t *handle, const alpm_list_t *urls,
 	if(payloads) {
 		event.type = ALPM_EVENT_PKG_RETRIEVE_START;
 		event.pkg_retrieve.num = alpm_list_count(payloads);
+		event.pkg_retrieve.total_size = 0;
 		EVENT(handle, &event);
 		if(_alpm_download(handle, payloads, cachedir) == -1) {
 			_alpm_log(handle, ALPM_LOG_WARNING, _("failed to retrieve some files\n"));
diff --git a/lib/libalpm/sync.c b/lib/libalpm/sync.c
index 294a4b93..d68df54e 100644
--- a/lib/libalpm/sync.c
+++ b/lib/libalpm/sync.c
@@ -833,6 +833,8 @@ static int download_files(alpm_handle_t *handle)
 		}
 
 		event.type = ALPM_EVENT_PKG_RETRIEVE_START;
+		event.pkg_retrieve.total_size = 0;
+		event.pkg_retrieve.num = 0;
 
 		/* sum up the number of packages to download and its total size */
 		for(i = files; i; i = i->next) {
